packages/ra-core/src/form/withDefaultValue.js

```diff
@@ -0,0 +1,57 @@
+import { Component, createElement } from 'react';
+import PropTypes from 'prop-types';
+
+import { connect } from 'react-redux';
+import { initializeForm as initializeFormAction } from '../actions/formActions';
+
+export class DefaultValue extends Component {
+    static propTypes = {
+        component: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+        defaultValue: PropTypes.any,
+        initializeForm: PropTypes.func.isRequired,
+        input: PropTypes.object,
+        source: PropTypes.string,
+        validate: PropTypes.oneOfType([PropTypes.func, PropTypes.array]),
+    };
+
+    componentDidMount() {
+        const { defaultValue, input, initializeForm, source } = this.props;
+        if (typeof defaultValue === 'undefined' || input) {
+            return;
+        }
+        initializeForm({
+            [source]:
+                typeof defaultValue === 'function'
+                    ? defaultValue()
+                    : defaultValue,
+        });
+    }
+
+    componentDidUpdate(prevProps) {
```

Comment:
why use `componentDidUpdate` instead of the previous `componentWillReceiveProps`?

Reply:
because `componentWillReceiveProps` is deprecated. See https://reactjs.org/blog/2018/03/29/react-v-16-3.html